flow/layers/physical_shape_layer.cc

```diff
@@ -78,30 +87,64 @@ void PhysicalShapeLayer::UpdateScene(SceneUpdateContext& context) {
   FML_DCHECK(needs_system_composite());
   TRACE_EVENT0("flutter", "PhysicalShapeLayer::UpdateScene");
 
-  // Retained rendering: speedup by reusing a retained entity node if possible.
-  // When an entity node is reused, no paint layer is added to the frame so we
-  // won't call PhysicalShapeLayer::Paint.
-  LayerRasterCacheKey key(unique_id(), context.Matrix());
-  if (context.HasRetainedNode(key)) {
-    TRACE_EVENT_INSTANT0("flutter", "retained layer cache hit");
-    const scenic::EntityNode& retained_node = context.GetRetainedNode(key);
-    FML_DCHECK(context.top_entity());
-    FML_DCHECK(retained_node.session() == context.session());
-    context.top_entity()->entity_node().AddChild(retained_node);
-    return;
-  }
+  // If there is embedded Fuchsia content in the scene (a ChildSceneLayer),
+  // PhysicalShapeLayers that appear above the embedded content will be turned
+  // into their own Scenic layers.
+  if (child_layer_exists_below_) {
+    // Reset paint bounds, because shadows are not rendered in this case
```

Comment:
Are you sure about this part?  There is no comment (or changes at all for that matter) in Paint()

Reply:
Actually, after reading forward I'm even more sure.  If you follow my other comment below in this function, you can remove this

Reply:
Look for "// TODO: Re-enable shadow drawing here." in Pain().

I think our offline discussion clarified why we need to do a check for child_layer_exists_below_; keeping it. Let me know if that wasn't the case.